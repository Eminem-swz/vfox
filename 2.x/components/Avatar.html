<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Avatar 头像 | Vfox for Vue3</title>
    <meta name="generator" content="VuePress 1.7.1">
    
    <meta name="description" content="移动端 Vue 组件库">
    
    <link rel="preload" href="/2.x/docs/assets/css/0.styles.508af179.css" as="style"><link rel="preload" href="/2.x/docs/assets/js/app.ad39ab2a.js" as="script"><link rel="preload" href="/2.x/docs/assets/js/2.bacaa9b2.js" as="script"><link rel="preload" href="/2.x/docs/assets/js/22.317bec75.js" as="script"><link rel="prefetch" href="/2.x/docs/assets/js/10.4d3031aa.js"><link rel="prefetch" href="/2.x/docs/assets/js/11.b8e9d0e5.js"><link rel="prefetch" href="/2.x/docs/assets/js/12.b4800caa.js"><link rel="prefetch" href="/2.x/docs/assets/js/13.835e5e58.js"><link rel="prefetch" href="/2.x/docs/assets/js/14.67692712.js"><link rel="prefetch" href="/2.x/docs/assets/js/15.52f7507b.js"><link rel="prefetch" href="/2.x/docs/assets/js/16.3706259a.js"><link rel="prefetch" href="/2.x/docs/assets/js/17.71052dd5.js"><link rel="prefetch" href="/2.x/docs/assets/js/18.cad7cf19.js"><link rel="prefetch" href="/2.x/docs/assets/js/19.6ffb4022.js"><link rel="prefetch" href="/2.x/docs/assets/js/20.7710a40e.js"><link rel="prefetch" href="/2.x/docs/assets/js/21.028c2ae2.js"><link rel="prefetch" href="/2.x/docs/assets/js/23.4d65baa5.js"><link rel="prefetch" href="/2.x/docs/assets/js/24.8dda71b0.js"><link rel="prefetch" href="/2.x/docs/assets/js/25.8200d57a.js"><link rel="prefetch" href="/2.x/docs/assets/js/26.0a2e60cc.js"><link rel="prefetch" href="/2.x/docs/assets/js/27.021711b6.js"><link rel="prefetch" href="/2.x/docs/assets/js/28.91e05e61.js"><link rel="prefetch" href="/2.x/docs/assets/js/29.d6ae50c5.js"><link rel="prefetch" href="/2.x/docs/assets/js/3.d693210d.js"><link rel="prefetch" href="/2.x/docs/assets/js/30.bdc3b9c4.js"><link rel="prefetch" href="/2.x/docs/assets/js/31.f994a029.js"><link rel="prefetch" href="/2.x/docs/assets/js/32.d4803146.js"><link rel="prefetch" href="/2.x/docs/assets/js/33.f969585f.js"><link rel="prefetch" href="/2.x/docs/assets/js/34.56ddc507.js"><link rel="prefetch" href="/2.x/docs/assets/js/35.8a3d4475.js"><link rel="prefetch" href="/2.x/docs/assets/js/36.165d30e9.js"><link rel="prefetch" href="/2.x/docs/assets/js/37.308813cb.js"><link rel="prefetch" href="/2.x/docs/assets/js/38.a3d7856b.js"><link rel="prefetch" href="/2.x/docs/assets/js/39.fa65cd15.js"><link rel="prefetch" href="/2.x/docs/assets/js/4.c2adc090.js"><link rel="prefetch" href="/2.x/docs/assets/js/40.10054c93.js"><link rel="prefetch" href="/2.x/docs/assets/js/41.0fd1a91c.js"><link rel="prefetch" href="/2.x/docs/assets/js/42.ba2bc92c.js"><link rel="prefetch" href="/2.x/docs/assets/js/43.f7e67e75.js"><link rel="prefetch" href="/2.x/docs/assets/js/44.cd76ef03.js"><link rel="prefetch" href="/2.x/docs/assets/js/45.f0ada67e.js"><link rel="prefetch" href="/2.x/docs/assets/js/46.cb03ae86.js"><link rel="prefetch" href="/2.x/docs/assets/js/47.c121548b.js"><link rel="prefetch" href="/2.x/docs/assets/js/48.3894e822.js"><link rel="prefetch" href="/2.x/docs/assets/js/49.fa698566.js"><link rel="prefetch" href="/2.x/docs/assets/js/5.8ae9a8f0.js"><link rel="prefetch" href="/2.x/docs/assets/js/50.abb41093.js"><link rel="prefetch" href="/2.x/docs/assets/js/51.6436ae56.js"><link rel="prefetch" href="/2.x/docs/assets/js/52.5ab6f279.js"><link rel="prefetch" href="/2.x/docs/assets/js/53.d8ddb489.js"><link rel="prefetch" href="/2.x/docs/assets/js/54.85e6a146.js"><link rel="prefetch" href="/2.x/docs/assets/js/55.c85faa47.js"><link rel="prefetch" href="/2.x/docs/assets/js/56.d11aeaeb.js"><link rel="prefetch" href="/2.x/docs/assets/js/57.ce8d3b01.js"><link rel="prefetch" href="/2.x/docs/assets/js/58.b136c83a.js"><link rel="prefetch" href="/2.x/docs/assets/js/59.b9a9f76c.js"><link rel="prefetch" href="/2.x/docs/assets/js/6.5ddd5f2e.js"><link rel="prefetch" href="/2.x/docs/assets/js/60.c51a8f0b.js"><link rel="prefetch" href="/2.x/docs/assets/js/61.38c7e6e2.js"><link rel="prefetch" href="/2.x/docs/assets/js/62.63203c78.js"><link rel="prefetch" href="/2.x/docs/assets/js/63.f95ec345.js"><link rel="prefetch" href="/2.x/docs/assets/js/64.ec0cb62a.js"><link rel="prefetch" href="/2.x/docs/assets/js/65.4b0e8cfa.js"><link rel="prefetch" href="/2.x/docs/assets/js/66.a1cf0143.js"><link rel="prefetch" href="/2.x/docs/assets/js/67.9edf1d05.js"><link rel="prefetch" href="/2.x/docs/assets/js/68.85776dac.js"><link rel="prefetch" href="/2.x/docs/assets/js/69.28fb2927.js"><link rel="prefetch" href="/2.x/docs/assets/js/7.03a52a57.js"><link rel="prefetch" href="/2.x/docs/assets/js/70.1b0b7dff.js"><link rel="prefetch" href="/2.x/docs/assets/js/71.c09dc408.js"><link rel="prefetch" href="/2.x/docs/assets/js/72.2489fdda.js"><link rel="prefetch" href="/2.x/docs/assets/js/73.37d54757.js"><link rel="prefetch" href="/2.x/docs/assets/js/74.7fe3e2b3.js"><link rel="prefetch" href="/2.x/docs/assets/js/75.f3fbbe42.js"><link rel="prefetch" href="/2.x/docs/assets/js/76.cbf8c9cf.js"><link rel="prefetch" href="/2.x/docs/assets/js/77.db10b7b0.js"><link rel="prefetch" href="/2.x/docs/assets/js/78.a6b2f90a.js"><link rel="prefetch" href="/2.x/docs/assets/js/79.bc633415.js"><link rel="prefetch" href="/2.x/docs/assets/js/8.87cfae1a.js"><link rel="prefetch" href="/2.x/docs/assets/js/80.99fc56d1.js"><link rel="prefetch" href="/2.x/docs/assets/js/81.abe16ca9.js"><link rel="prefetch" href="/2.x/docs/assets/js/82.6b5c642b.js"><link rel="prefetch" href="/2.x/docs/assets/js/83.3c0fc69c.js"><link rel="prefetch" href="/2.x/docs/assets/js/84.8911102d.js"><link rel="prefetch" href="/2.x/docs/assets/js/85.b4f255ae.js"><link rel="prefetch" href="/2.x/docs/assets/js/86.0558f824.js"><link rel="prefetch" href="/2.x/docs/assets/js/87.5369a89a.js"><link rel="prefetch" href="/2.x/docs/assets/js/88.b7a26cef.js"><link rel="prefetch" href="/2.x/docs/assets/js/89.dd29dc01.js"><link rel="prefetch" href="/2.x/docs/assets/js/9.c72cbeff.js"><link rel="prefetch" href="/2.x/docs/assets/js/90.64005830.js"><link rel="prefetch" href="/2.x/docs/assets/js/91.db95a78f.js">
    <link rel="stylesheet" href="/2.x/docs/assets/css/0.styles.508af179.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/2.x/docs/" class="home-link router-link-active"><img src="/2.x/docs/assets/images/logo.png" alt="Vfox for Vue3" class="logo"> <span class="site-name can-hide">Vfox for Vue3</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/2.x/docs/" class="nav-link">
  指南
</a></div><div class="nav-item"><a href="/2.x/docs/design/" class="nav-link">
  设计
</a></div><div class="nav-item"><a href="/2.x/docs/components/Button.html" class="nav-link">
  组件
</a></div><div class="nav-item"><a href="/2.x/docs/apis/LocalStorage.html" class="nav-link">
  API
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="选择版本" class="dropdown-title"><span class="title">选择版本</span> <span class="arrow down"></span></button> <button type="button" aria-label="选择版本" class="mobile-dropdown-title"><span class="title">选择版本</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  1.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-item"><!----> <a href="/2.x/docs/" class="nav-link">
  2.x
</a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/2.x/docs/" class="nav-link">
  指南
</a></div><div class="nav-item"><a href="/2.x/docs/design/" class="nav-link">
  设计
</a></div><div class="nav-item"><a href="/2.x/docs/components/Button.html" class="nav-link">
  组件
</a></div><div class="nav-item"><a href="/2.x/docs/apis/LocalStorage.html" class="nav-link">
  API
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="选择版本" class="dropdown-title"><span class="title">选择版本</span> <span class="arrow down"></span></button> <button type="button" aria-label="选择版本" class="mobile-dropdown-title"><span class="title">选择版本</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  1.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-item"><!----> <a href="/2.x/docs/" class="nav-link">
  2.x
</a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/2.x/docs/" aria-current="page" class="sidebar-link">指南</a></li><li><a href="/2.x/docs/design/" class="sidebar-link">设计</a></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>基础</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/2.x/docs/components/Button.html" class="sidebar-link">Button 按钮</a></li><li><a href="/2.x/docs/components/Icon.html" class="sidebar-link">Icon 图标</a></li><li><a href="/2.x/docs/components/Image.html" class="sidebar-link">Image 图片</a></li><li><a href="/2.x/docs/components/ScrollView.html" class="sidebar-link">ScrollView 滚动区</a></li><li><a href="/2.x/docs/components/Drawer.html" class="sidebar-link">Drawer 抽屉</a></li><li><a href="/2.x/docs/components/Modal.html" class="sidebar-link">Modal 模态框</a></li><li><a href="/2.x/docs/components/Popover.html" class="sidebar-link">Popover 气泡</a></li><li><a href="/2.x/docs/components/Dropdown.html" class="sidebar-link">Dropdown 下拉框</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>反馈</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/2.x/docs/components/Toast.html" class="sidebar-link">Toast 轻提示</a></li><li><a href="/2.x/docs/components/Dialog.html" class="sidebar-link">Dialog 模态对话框</a></li><li><a href="/2.x/docs/components/Notify.html" class="sidebar-link">Notify 消息提示</a></li><li><a href="/2.x/docs/components/ScrollView.html" class="sidebar-link">ScrollView 滚动区</a></li><li><a href="/2.x/docs/components/ActionSheet.html" class="sidebar-link">ActionSheet</a></li><li><a href="/2.x/docs/components/PopDialog.html" class="sidebar-link">PopDialog 气泡对话框</a></li><li><a href="/2.x/docs/components/SwipeCell.html" class="sidebar-link">SwipeCell 滑动单元格</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>数据输入</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/2.x/docs/components/Form.html" class="sidebar-link">Form 表单</a></li><li><a href="/2.x/docs/components/Input.html" class="sidebar-link">Input 输入框</a></li><li><a href="/2.x/docs/components/Radio.html" class="sidebar-link">Radio 单选项</a></li><li><a href="/2.x/docs/components/Checkbox.html" class="sidebar-link">Checkbox 多选项</a></li><li><a href="/2.x/docs/components/Slider.html" class="sidebar-link">Slider 滑块选择器</a></li><li><a href="/2.x/docs/components/Switch.html" class="sidebar-link">Switch 开关选择器</a></li><li><a href="/2.x/docs/components/Stepper.html" class="sidebar-link">Stepper 步进器</a></li><li><a href="/2.x/docs/components/Cascader.html" class="sidebar-link">Cascader 级联选择器</a></li><li><a href="/2.x/docs/components/Picker.html" class="sidebar-link">Picker 选择器</a></li><li><a href="/2.x/docs/components/Calendar.html" class="sidebar-link">Calendar 日历</a></li><li><a href="/2.x/docs/components/SearchBar.html" class="sidebar-link">SearchBar 搜索栏</a></li><li><a href="/2.x/docs/components/NumberKeyboard.html" class="sidebar-link">NumberKeyboard 数字键盘</a></li><li><a href="/2.x/docs/components/ImageUploader.html" class="sidebar-link">ImageUploader 图片上传</a></li><li><a href="/2.x/docs/components/DatePicker.html" class="sidebar-link">DatePicker 时间选择器</a></li><li><a href="/2.x/docs/components/Range.html" class="sidebar-link">Range 区间选择器</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>导航</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/2.x/docs/components/BackTop.html" class="sidebar-link">BackTop 返回顶部</a></li><li><a href="/2.x/docs/components/NavBar.html" class="sidebar-link">NavBar 导航栏</a></li><li><a href="/2.x/docs/components/PopMenu.html" class="sidebar-link">PopMenu 气泡菜单</a></li><li><a href="/2.x/docs/components/ScrollView.html" class="sidebar-link">ScrollView 滚动区</a></li><li><a href="/2.x/docs/components/Tab.html" class="sidebar-link">Tab 标签列表</a></li><li><a href="/2.x/docs/components/SideTab.html" class="sidebar-link">SideTab 侧边标签栏</a></li><li><a href="/2.x/docs/components/TabBar.html" class="sidebar-link">TabBar 标签列</a></li><li><a href="/2.x/docs/components/TabView.html" class="sidebar-link">TabView 标签页</a></li><li><a href="/2.x/docs/components/Sticky.html" class="sidebar-link">Sticky 粘性布局</a></li><li><a href="/2.x/docs/components/ScrollTab.html" class="sidebar-link">ScrollTab 标签滚动布局</a></li><li><a href="/2.x/docs/components/IndexView.html" class="sidebar-link">IndexView 索引栏</a></li><li><a href="/2.x/docs/components/Fixed.html" class="sidebar-link">Fixed 固定布局</a></li><li><a href="/2.x/docs/components/Pagination.html" class="sidebar-link">Pagination 分页</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>展示</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/2.x/docs/components/Layout.html" class="sidebar-link">Layout 布局</a></li><li><a href="/2.x/docs/components/Cell.html" class="sidebar-link">Cell 单元格</a></li><li><a href="/2.x/docs/components/Group.html" class="sidebar-link">Group 分组</a></li><li><a href="/2.x/docs/components/Collapse.html" class="sidebar-link">Collapse 折叠面板</a></li><li><a href="/2.x/docs/components/Empty.html" class="sidebar-link">Empty 空状态</a></li><li><a href="/2.x/docs/components/Skeleton.html" class="sidebar-link">Skeleton 骨架屏</a></li><li><a href="/2.x/docs/components/Swiper.html" class="sidebar-link">Swiper 轮播</a></li><li><a href="/2.x/docs/components/ImagePreview.html" class="sidebar-link">ImagePreview 图片预览</a></li><li><a href="/2.x/docs/components/FlatList.html" class="sidebar-link">FlatList 回收列表</a></li><li><a href="/2.x/docs/components/NoticeBar.html" class="sidebar-link">NoticeBar 通告栏</a></li><li><a href="/2.x/docs/components/Divider.html" class="sidebar-link">Divider 分割线</a></li><li><a href="/2.x/docs/components/Badge.html" class="sidebar-link">Badge 徽标数</a></li><li><a href="/2.x/docs/components/Progress.html" class="sidebar-link">Progress 进度条</a></li><li><a href="/2.x/docs/components/Tag.html" class="sidebar-link">Tag 标签</a></li><li><a href="/2.x/docs/components/Order.html" class="sidebar-link">Order 调整排序布局</a></li><li><a href="/2.x/docs/components/CountDown.html" class="sidebar-link">CountDown 倒计时</a></li><li><a href="/2.x/docs/components/Steps.html" class="sidebar-link">Steps 步骤条</a></li><li><a href="/2.x/docs/components/Price.html" class="sidebar-link">Price 价格</a></li><li><a href="/2.x/docs/components/Avatar.html" aria-current="page" class="active sidebar-link">Avatar 头像</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#avatar-props" class="sidebar-link">Avatar Props</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#size-的合法值" class="sidebar-link">size 的合法值</a></li><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#color-style-的合法值" class="sidebar-link">color-style 的合法值</a></li><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#shape-的合法值" class="sidebar-link">shape 的合法值</a></li></ul></li><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#avatar-slots" class="sidebar-link">Avatar Slots</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#默认-default" class="sidebar-link">默认（#default）</a></li></ul></li><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#avatargroup-头像组" class="sidebar-link">AvatarGroup 头像组</a></li><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#avatargroup-props" class="sidebar-link">AvatarGroup Props</a></li><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#avatargroup-slots" class="sidebar-link">AvatarGroup Slots</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/2.x/docs/components/Avatar.html#默认-default-2" class="sidebar-link">默认（#default）</a></li></ul></li></ul></li><li><a href="/2.x/docs/components/QrCode.html" class="sidebar-link">QrCode 二维码</a></li><li><a href="/2.x/docs/components/Timeline.html" class="sidebar-link">Timeline 步骤条</a></li><li><a href="/2.x/docs/components/Stopwatch.html" class="sidebar-link">Stopwatch 秒表</a></li><li><a href="/2.x/docs/components/Result.html" class="sidebar-link">Result 结果</a></li><li><a href="/2.x/docs/components/LoadMore.html" class="sidebar-link">LoadMore 页底</a></li><li><a href="/2.x/docs/components/TimeAgo.html" class="sidebar-link">TimeAgo 相对时间</a></li><li><a href="/2.x/docs/components/CountUp.html" class="sidebar-link">CountUp 数字动画</a></li><li><a href="/2.x/docs/components/CircleProgress.html" class="sidebar-link">CircleProgress 环形进度条</a></li><li><a href="/2.x/docs/components/ActivityIndicator.html" class="sidebar-link">ActivityIndicator 活动指示器</a></li></ul></section></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading"><span>其他</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/2.x/docs/components/Copy.html" class="sidebar-link">Copy 复制</a></li></ul></section></li><li><a href="/2.x/docs/apis/LocalStorage.html" class="sidebar-link">API</a></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="avatar-头像"><a href="#avatar-头像" class="header-anchor">#</a> Avatar 头像</h1> <h2 id="avatar-props"><a href="#avatar-props" class="header-anchor">#</a> Avatar Props</h2> <table><thead><tr><th>属性</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>src</td> <td>string</td> <td></td> <td>否</td> <td>图片连接，显示自定义图片</td></tr> <tr><td>size</td> <td>string/number</td> <td>'middle'</td> <td>否</td> <td>头像尺寸，处除了预设值，还支持传入 number</td></tr> <tr><td>color-style</td> <td>string</td> <td></td> <td>否</td> <td>色彩风格</td></tr> <tr><td>shape</td> <td>string</td> <td>'circle'</td> <td>否</td> <td>头像形状</td></tr> <tr><td>badge</td> <td>string/number/object</td> <td></td> <td>否</td> <td>徽标，使用 <a href="/2.x/docs/components/Badge.html">Badge</a> 组件，可传入一个基于 Badge props 的对象</td></tr> <tr><td>gender</td> <td>string</td> <td></td> <td>否</td> <td>显示男女角标，优先级高于 badge</td></tr></tbody></table> <h3 id="size-的合法值"><a href="#size-的合法值" class="header-anchor">#</a> size 的合法值</h3> <table><thead><tr><th>值</th> <th>说明</th></tr></thead> <tbody><tr><td>large</td> <td>大尺寸，宽高 64px，文字 32px，图标 43px</td></tr> <tr><td>middle</td> <td>默认值，中尺寸，宽高 48px，文字 24px，图标 32px</td></tr> <tr><td>large</td> <td>小尺寸，宽高 36px，文字 18px，图标 21px</td></tr></tbody></table> <h3 id="color-style-的合法值"><a href="#color-style-的合法值" class="header-anchor">#</a> color-style 的合法值</h3> <table><thead><tr><th>值</th> <th>说明</th></tr></thead> <tbody><tr><td>DustRed</td> <td><font color="#ff4d4f">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>Volcano</td> <td><font color="#ff7a45">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>SunsetOrange</td> <td><font color="#fa8c16">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>CalendulaGold</td> <td><font color="#faad14">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>SunriseYellow</td> <td><font color="#fadb14">背景色</font> <font color="#262626">前景色</font></td></tr> <tr><td>Lime</td> <td><font color="#a0d911">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>PolarGreen</td> <td><font color="#52c41a">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>Cyan</td> <td><font color="#36cfc9">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>DaybreakBlue</td> <td><font color="#1890ff">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>GeekBlue</td> <td><font color="#597ef7">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>GoldenPurple</td> <td><font color="#9254de">背景色</font> <font color="#ffffff">前景色</font></td></tr> <tr><td>Magenta</td> <td><font color="#f759ab">背景色</font> <font color="#ffffff">前景色</font></td></tr></tbody></table> <h3 id="shape-的合法值"><a href="#shape-的合法值" class="header-anchor">#</a> shape 的合法值</h3> <table><thead><tr><th>值</th> <th>说明</th></tr></thead> <tbody><tr><td>circle</td> <td>圆形头像</td></tr> <tr><td>square</td> <td>圆角方形头像</td></tr></tbody></table> <h2 id="avatar-slots"><a href="#avatar-slots" class="header-anchor">#</a> Avatar Slots</h2> <h3 id="默认-default"><a href="#默认-default" class="header-anchor">#</a> 默认（#default）</h3> <div class="language- extra-class"><pre class="language-text"><code>&lt;fx-avatar&gt;曹&lt;/fx-avatar&gt;
</code></pre></div><h2 id="avatargroup-头像组"><a href="#avatargroup-头像组" class="header-anchor">#</a> AvatarGroup 头像组</h2> <h2 id="avatargroup-props"><a href="#avatargroup-props" class="header-anchor">#</a> AvatarGroup Props</h2> <table><thead><tr><th>属性</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>size</td> <td>string/number</td> <td>'middle'</td> <td>否</td> <td>组内头像尺寸，处除了预设值，还支持传入 number</td></tr> <tr><td>total-count</td> <td>number</td> <td></td> <td>否</td> <td>尾部显示头像数，如果数值过高，会对数值简化处理，如 1.1w</td></tr> <tr><td>count-color-style</td> <td>string</td> <td></td> <td>否</td> <td>头像数色彩风格，同 <a href="/2.x/docs/components/Avatar.html#color-style-的合法值">Avatar</a></td></tr></tbody></table> <h2 id="avatargroup-slots"><a href="#avatargroup-slots" class="header-anchor">#</a> AvatarGroup Slots</h2> <h3 id="默认-default-2"><a href="#默认-default-2" class="header-anchor">#</a> 默认（#default）</h3> <p>注：其中只可放置 <a href="/2.x/docs/components/Collapse.html#avatar-头像">Avatar</a> 组件，否则会导致未定义的行为。</p> <div class="language- extra-class"><pre class="language-text"><code>&lt;fx-avatar-group&gt;
  &lt;fx-avatar src=&quot;https://cdn.fox2.cn/vfox/avatar/5.png&quot; /&gt;
  &lt;fx-avatar src=&quot;https://cdn.fox2.cn/vfox/avatar/6.png&quot; /&gt;
  &lt;fx-avatar src=&quot;https://cdn.fox2.cn/vfox/avatar/7.png&quot; /&gt;
&lt;/fx-avatar-group&gt;
</code></pre></div></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/2.x/docs/components/Price.html" class="prev">
        Price 价格
      </a></span> <span class="next"><a href="/2.x/docs/components/QrCode.html">
        QrCode 二维码
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/2.x/docs/assets/js/app.ad39ab2a.js" defer></script><script src="/2.x/docs/assets/js/2.bacaa9b2.js" defer></script><script src="/2.x/docs/assets/js/22.317bec75.js" defer></script>
  </body>
</html>
