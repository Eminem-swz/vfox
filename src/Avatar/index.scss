@import '../style/var.scss';

.fx-avatar {
  --avatar-size: 48px;
  --avatar-font-size: 24px;
  --avatar-color: #{$font-color};

  background-color: $background-color;
  color: #fff;
  width: var(--avatar-size);
  height: var(--avatar-size);
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  font-size: var(--avatar-font-size);
  color: var(--avatar-color);
  box-sizing: border-box;

  .fx-image {
    width: 100%;
    height: 100%;
    border-radius: 50%;

    .fx-icon {
      --size: 1.333em;
      --color: var(--avatar-color);
    }
  }

  &.shape--square {
    border-radius: 4px;

    .fx-image {
      border-radius: 4px;
    }
  }

  &.size--large {
    --avatar-size: 64px;
    --avatar-font-size: 32px;
  }

  &.size--small {
    --avatar-size: 36px;
    --avatar-font-size: 18px;
  }

  &.color-- {
    &DustRed {
      background-color: #ff4d4f;
      --avatar-color: #fff;
    }
    &Volcano {
      background-color: #ff7a45;
      --avatar-color: #fff;
    }
    &SunsetOrange {
      background-color: #fa8c16;
      --avatar-color: #fff;
    }
    &CalendulaGold {
      background-color: #faad14;
      --avatar-color: #fff;
    }
    &SunriseYellow {
      background-color: #fadb14;
      --avatar-color: #{$title-color};
    }
    &Lime {
      background-color: #a0d911;
      --avatar-color: #fff;
    }
    &PolarGreen {
      background-color: #52c41a;
      --avatar-color: #fff;
    }
    &Cyan {
      background-color: #36cfc9;
      --avatar-color: #fff;
    }
    &DaybreakBlue {
      background-color: #1890ff;
      --avatar-color: #fff;
    }
    &GeekBlue {
      background-color: #597ef7;
      --avatar-color: #fff;
    }
    &GoldenPurple {
      background-color: #9254de;
      --avatar-color: #fff;
    }
    &Magenta {
      background-color: #f759ab;
      --avatar-color: #fff;
    }
  }
}

.fx-avatar-group {
  > .fx-avatar {
    border: 1px solid #fff;
    margin-left: -0.5em;

    &:first-child {
      margin-left: 0;
    }
  }

  &_count {
    &-number {
      font-size: 0.667em;

      &.size--3 {
        font-size: 0.611em;
      }

      &.size--3 {
        font-size: 0.556em;
      }

      &.size--4 {
        font-size: 0.556em;
      }

      &.size--5 {
        font-size: 0.5em;
      }
    }
  }
}
