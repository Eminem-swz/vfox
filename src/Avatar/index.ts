import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import Avatar from './Avatar.vue'

const _Avatar: SFCWithInstall<typeof Avatar> = Object.assign(Avatar, {
  install: function (app: App) {
    app.component(Avatar.name, Avatar)
  }
})

export default _Avatar
