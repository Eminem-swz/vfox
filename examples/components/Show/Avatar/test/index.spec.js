/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import Avatar from '@/Avatar'

describe('Avatar', () => {
  test('should render default correctly', () => {
    const wrapper = shallowMount(Avatar, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
