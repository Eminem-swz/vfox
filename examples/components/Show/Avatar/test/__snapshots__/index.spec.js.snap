// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render default correctly 1`] = `<fx-badge-stub class="fx-avatar size--middle shape--circle color--default" offset="0,0"></fx-badge-stub>`;
